File: model-optimizer/extensions/front/caffe/tanh.py
# Copyright (C) 2018-2021 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

from extensions.ops.activation_ops import Tanh
from mo.front.extractor import FrontExtractorOp


class TanhFrontExtractor(FrontExtractorOp):
    op = 'Tanh'
    enabled = True

    @classmethod
    def extract(cls, node):
        Tanh.update_node_stat(node)
        return cls.enabled

File: ngraph/frontend/paddlepaddle/src/op/tanh.cpp
// Copyright (C) 2018-2021 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include <node_context.hpp>

#include "default_opset.hpp"

namespace ngraph {
namespace frontend {
namespace pdpd {
namespace op {
NamedOutputs tanh(const NodeContext& node) {
    const auto x = node.get_ng_input("X");

    return node.default_single_output_mapping({std::make_shared<default_opset::Tanh>(x)}, {"Out"});
}

}  // namespace op
}  // namespace pdpd
}  // namespace frontend
}  // namespace ngraph
